File: polyglot/runtime_template_resolver/mjs/src/compiler.ts
import { logger } from './logger.js';
import { TemplateResolver } from './resolver.js';
import { ITemplateResolver } from './interfaces.js';

const log = logger.create('runtime-template-resolver', 'compiler.ts');
const textResolver: ITemplateResolver = new TemplateResolver();

export function compile(template: string): (context: Record<string, unknown>) => string {
    log.debug('Compiling template', { template });
    return (context: Record<string, unknown>) => {
        return textResolver.resolve(template, context);
    };
}

File: polyglot/runtime_template_resolver/py/src/runtime_template_resolver/compiler.py
from typing import Callable, Any, Dict
from .logger import logger
from .resolver import TemplateResolver

log = logger.create("runtime_template_resolver", __file__)
_resolver = TemplateResolver()

def compile(template: str) -> Callable[[Dict[str, Any]], str]:
    log.debug("Compiling template", template=template)
    def compiled(context: Dict[str, Any]) -> str:
        return _resolver.resolve(template, context)
    return compiled
